app/digitizingcontroller.cpp

```diff
@@ -141,10 +153,17 @@ void DigitizingController::onPositionChanged()
   if ( !mPositionKit->hasPosition() )
     return;
 
+
   QgsPoint point = mPositionKit->position();
   fixZ(&point);
 
-  mRecordedPoints.append( point );
+  if (mDigitizingPeriodTimer.isActive()) {
+      mRecordedPoints.last().setX(point.x());
+      mRecordedPoints.last().setY(point.y());
+  } else {
+      mRecordedPoints.append( point );
+      mDigitizingPeriodTimer.start( mDigitizingPeriod * 1000 );
+  }
```

Comment:
Does this work correctly? I am worried that if the QTimer gets deactivated around the time of new gps position coming, we may be getting inconsistent recording (it could be +/- one second). I think it would be better if we just kept QDateTime of the last position (from QDateTime::currentDateTime()) and then check if secsTo() is >= than our interval. The QTimer class is more suited for cases when we want to trigger an action on its timeout() signal...